lib/react_test_utils.dart

```diff
@@ -31,6 +31,17 @@ _getNestedJsObject(
   return object;
 }
 
+/// Returns [component] if it is already a [JsObject], and converts [component] if
+/// it is an [Element]. If [component] is neither a [JsObject] or [Element], throws an
+/// [ArgumentError].
```

Comment:
I think this should include some sort of explanation as to why you would want to do this.

A shortened version what you have in your PR description would be perfect IMO:
> Since React no longer uses the same type of object for primitive components as composite components, Dart converts the React objects for primitive components to Elements automatically. This is problematic in some cases - primarily the test utility methods that return JsObjects, and render, which also needs to return a JsObject. I'm handling this by converting the Element back to a JsObject - it still has the keys you'd expect on a ReactElement JsObject (_reactInternalComponent, props, tagName). I'm open to other ideas about handling this change, though.